qiita_db/test/test_user.py

```diff
@@ -224,38 +219,85 @@ def test_set_info_bad_info(self):
             self.user.info = self.userinfo
 
     def test_default_analysis(self):
-        obs = self.user.default_analysis
-        self.assertEqual(obs, 4)
+        try:
+            qiita_config.portal = "QIITA"
+            obs = self.user.default_analysis
+            self.assertEqual(obs, 4)
+
+            qiita_config.portal = "EMP"
+            obs = self.user.default_analysis
+            self.assertEqual(obs, 8)
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_user_studies(self):
         user = User('test@foo.bar')
-        self.assertEqual(user.user_studies, {1})
+        try:
+            qiita_config.portal = "QIITA"
+            self.assertEqual(user.user_studies, {1})
+
+            qiita_config.portal = "EMP"
+            self.assertEqual(user.user_studies, set())
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_shared_studies(self):
         user = User('shared@foo.bar')
-        self.assertEqual(user.shared_studies, {1})
+        try:
+            qiita_config.portal = "QIITA"
+            self.assertEqual(user.shared_studies, {1})
+
+            qiita_config.portal = "EMP"
+            self.assertEqual(user.shared_studies, set())
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_private_analyses(self):
-        self.assertEqual(self.user.private_analyses, set([]))
+        user = User('test@foo.bar')
+        try:
+            qiita_config.portal = "QIITA"
+            self.assertEqual(user.private_analyses, set([1, 2]))
+
+            qiita_config.portal = "EMP"
+            self.assertEqual(user.private_analyses, set())
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_shared_analyses(self):
-        self.assertEqual(self.user.shared_analyses, set([]))
+        user = User('shared@foo.bar')
+        try:
```

Comment:
Given that there are multiple functions where this needs to be done, can you add it to the tearDown function? That is the whole idea of that tearDown function. Also, it doesn't hurt if it is executed always. and make the tests more readable.